src/scriptdev2/scripts/eastern_kingdoms/naxxramas/boss_anubrekhan.cpp

```diff
@@ -132,15 +128,24 @@ struct boss_anubrekhanAI : public ScriptedAI
         DoCastSpellIfCan(m_creature, SPELL_DOUBLE_ATTACK, CAST_TRIGGERED | CAST_AURA_NOT_PRESENT);
     }
 
-    void MoveInLineOfSight(Unit* pWho) override
+    void OnAnubDoorGo (Player* pPlayer, Creature* pCreature)
     {
-        if (!m_bHasTaunted && pWho->GetTypeId() == TYPEID_PLAYER && m_creature->IsWithinDistInMap(pWho, 110.0f) && m_creature->IsWithinLOSInMap(pWho))
+        if (GO_ARAC_ANUB_DOOR == GO_STATE_ACTIVE)
```

Comment:
The reasoning behind it is that we can just hook it to an event instead of having to do distance/LoS checks. I thought it would also look cleaner and/or make the event smoother.